### PR TITLE
No exit when union_helper not found

### PR DESCRIPTION
**Pull request by angelhof**
This PR fixes try to not exit when unionfs/mergerfs are not found, but just produce warning messages.

Addresses: #96

### CODE REVIEW

File: test/run_tests.sh [hunk 1]
```diff
@@ -394,6 +394,31 @@ test_dev()
...
+test_echo_no_unionfs_mergerfs()
+{
+    local try_workspace=$1
+    cd "$try_workspace/"
+
+    ## Create a new /bin and /usr/bin without mergerfs and unionfs 
+    new_bin_dir=$(mktemp -d)
+    cp -r /bin "$new_bin_dir/bin"
+    mkdir "$new_bin_dir/usr"
+    cp -r /usr/bin "$new_bin_dir/usr/bin"
```
Comment (mgree) [inline test/run_tests.sh:407]: Maybe better to symlink everything (excluding these files) than to copy them?

```
$ du -h /bin /usr/bin
0	/bin
1.1G	/usr/bin
```

**Review by ericzty — COMMENTED** [review]
Test cases worked on machines with mergerfs.
`try` now behaves correctly with machines nested mount and no unionfs/mergerfs.

```
eric@try-test-lvm:/srv$ /home/eric/try/try echo hi
try(/tmp/tmp.dRGKQR7sz5): Warning: Failed mounting /home as an overlay and mergerfs or unionfs not set and could not be found, see "/tmp/tmp.5rnwzXMHdm"
try(/tmp/tmp.dRGKQR7sz5): Warning: Failed mounting /run as an overlay and mergerfs or unionfs not set and could not be found, see "/tmp/tmp.5rnwzXMHdm"
try(/tmp/tmp.dRGKQR7sz5): Warning: Failed mounting /sys as an overlay and mergerfs or unionfs not set and could not be found, see "/tmp/tmp.5rnwzXMHdm"
try(/tmp/tmp.dRGKQR7sz5): Warning: Failed mounting /home as an overlay and mergerfs or unionfs not set and could not be found, see "/tmp/tmp.5rnwzXMHdm"
hi
```
home is repeated here
* one re-run was caused by /home being a mount(listed in findmnt)
* the other re-run is caused by the initial mounting of /home failing because it is a nested mount

Is this something that we want to look into or is this fine?

**Review by mgree — CHANGES_REQUESTED** [review]
I think the `cp -R` is going to be really expensive, so maybe symlinking instead? Or mark this test as `CI` only (i.e., the `CI` variable needs to be set).

File: test/run_tests.sh (hunk 1 above)
Comment (ericzty) [reply test/run_tests.sh:407]: I agree, on my machine the that test took a while.

**Comment by mgree** [comment]
> home is repeated here
> 
>     * one re-run was caused by /home being a mount(listed in findmnt)
> 
>     * the other re-run is caused by the initial mounting of /home failing because it is a nested mount
> 
> 
> Is this something that we want to look into or is this fine?

We should try to avoid duplicate work... and we should definitely avoid redundant warnings/output. Definitely worth fixing, especially if it's easy.

File: test/run_tests.sh (hunk 1 above)
Comment (angelhof) [reply test/run_tests.sh:407]: Sure let me figure out how to do that!

**Comment by angelhof** [comment]
> > home is repeated here
> > ```
> > * one re-run was caused by /home being a mount(listed in findmnt)
> > 
> > * the other re-run is caused by the initial mounting of /home failing because it is a nested mount
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Is this something that we want to look into or is this fine?
> 
> We should try to avoid duplicate work... and we should definitely avoid redundant warnings/output. Definitely worth fixing, especially if it's easy.

I think this is caused by `for mountpoint in /* $(findmnt --real -r -o target -n)` which lists both all top level directories and then all mounts. This is an issue that we figured out we need to solve indeed (but it doesn't have anything to do with my PR).

**Comment by ericzty** [comment]
On Thu Jul 6, 2023 at 4:57 PM EDT, Konstantinos Kallas wrote:
> I think this is caused by `for mountpoint in /* $(findmnt --real -r -o target -n)` which lists both all top level directories and then all mounts. This is an issue that we figured out we need to solve indeed (but it doesn't have anything to do with my PR). 
Perhaps we should move this to a separate issue and handle it outside of
this PR.

**Comment by angelhof** [comment]
I think the repeating of home also has to do with the first part of this issue (https://github.com/binpash/try/issues/104)

File: test/run_tests.sh (hunk 1 above)
Comment (angelhof) [reply test/run_tests.sh]: Done, using `-s` flag with `cp`.

**Comment by ericzty** [comment]
I think it is related to that, but the solution for the first part of
the issue is to make separate tempdirs to store the helper directories
for the nested mounts.

For this we want to make sure we only try to mount them once, probably
by removing the duplicates from `/*` and `findmnt --real -r -o target
-n`

Possible in the findmnt loop, we check if $mountpoint is in /*

**Review by ericzty — APPROVED** [review]
significant speed up! (my test vm is on a very old sas drive)

**Comment by angelhof** [comment]
OK, fixed the duplicate issue too @ericzty can you check you dont get the warning for home twice? @mgree could you check the new changes?

**Review by ericzty — APPROVED** [review]
```
eric@try-test-lvm:/srv$ /home/eric/try/try echo hi
try(/tmp/tmp.4B1SvKhn8b): Warning: Failed mounting /home as an overlay and mergerfs or unionfs not set and could not be found, see "/tmp/tmp.NJCocX3VGY"
try(/tmp/tmp.4B1SvKhn8b): Warning: Failed mounting /run as an overlay and mergerfs or unionfs not set and could not be found, see "/tmp/tmp.NJCocX3VGY"
try(/tmp/tmp.4B1SvKhn8b): Warning: Failed mounting /sys as an overlay and mergerfs or unionfs not set and could not be found, see "/tmp/tmp.NJCocX3VGY"
hi
```
:rocket:

**Pull request merged**